stock_quant_history/models/stock_quant_history_snapshot.py

```diff
@@ -0,0 +1,180 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+import logging
+from collections import defaultdict
+
+from odoo import _, api, fields, models
+from odoo.osv.expression import AND
+
+_logger = logging.getLogger(__name__)
+
+
+class DefaultDict(defaultdict):
+    def __missing__(self, key):
+        self[key] = self.default_factory(*key)
+        return self[key]
+
+
+class StockQuantHistorySnapshot(models.Model):
+    _name = "stock.quant.history.snapshot"
+    _description = "stock.quant.history generation configuration model"
+    _order = "inventory_date desc"
+
+    name = fields.Char(
+        compute="_compute_name",
+    )
+    stock_quant_history_ids = fields.One2many(
+        comodel_name="stock.quant.history",
+        inverse_name="snapshot_id",
+        string="Stock quant history",
+        help="Generated stock quant history for current snapshot settings.",
+    )
+    state = fields.Selection(
+        selection=[
+            ("draft", "Draft"),
+            ("generated", "Generated"),
+        ],
+        string="Status",
+        copy=False,
+        default="draft",
+        readonly=True,
+        required=True,
+    )
+
+    inventory_date = fields.Datetime(
+        string="Inventory date",
+        required=True,
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        help="The date used to create stock.quant.history as it was for the given date",
+    )
+    generated_date = fields.Datetime(
+        string="Generated date",
+        readonly=True,
+        copy=False,
+        help="Date when stock.quant.history line have been created.",
+    )
+    previous_snapshot_id = fields.Many2one(
+        comodel_name="stock.quant.history.snapshot",
+        string="Snapshot base",
+        readonly=True,
+        help="Base snapshot used to generate this snapshot",
+    )
+
+    @api.depends("inventory_date")
+    def _compute_name(self):
+        # Odoo enforce users to be linked to an active lang
+        lang = self.env["res.lang"]._lang_get(self.env.user.lang)
+        dt_format = lang.date_format + " " + lang.time_format
+
+        for rec in self:
+            rec.name = _("Snapshot %s") % (rec.inventory_date.strftime(dt_format))
+
+    def action_generate_stock_quant_history(self):
+        for snapshot in self:
+            snapshot._generate_stock_quant_history()
+
+    def _prepare_stock_move_line_filter(self, previous_quant_snapshot):
+        domain = [
+            ("state", "=", "done"),
+            ("date", "<=", self.inventory_date),
+        ]
+        if previous_quant_snapshot.exists():
+            domain = AND(
+                [domain, [("date", ">", previous_quant_snapshot.inventory_date)]]
+            )
+
+        return domain
+
+    @api.model
+    def _ignored_location_usage(self):
+        """If you overwrite or change this
+        list you'll probably want to regenerate all your
+        snapshots"""
+        return [
```

Comment:
**question:** is there a reason to provide these values by default? It's looks a bit to specific